internal/archiver/archiver.go

```diff
@@ -717,7 +717,7 @@ func resolveRelativeTargets(filesys fs.FS, targets []string) ([]string, error) {
 
 // SnapshotOptions collect attributes for a new snapshot.
 type SnapshotOptions struct {
-	Tags           restic.TagList
+	Tags           restic.TagLists
```

Comment:
I would keep the `TagList` in the archiver code and would use the `Flatten()` method in `cmd_backup.go`, line 685.